index.js

```diff
@@ -28,10 +37,184 @@ for (const file of commandFiles) {
+client.once('ready', () => { 
+	console.log('Ready !');	
+	sendNewClientMessage(1, ["Monday 9AM", "Wednesday 2PM", "Thursday 6PM"], 10, "Maths", "GCSE", 2, 1);
+	sendNewClientMessage(2, ["Monday 9AM", "Wednesday 2PM", "Thursday 6PM"], 10, "CS", "Uni", 1, 2);
+});
+
+/**
+ * Sends a message displaying a new client annoucement to tutor's discord channel.
+ * User has the possibility to select time slots via a button.
+ * @param {Array} availabilities 
+ * @param {number} money 
+ * @param {String} subject 
+ * @param {String} level 
+ * @param {number} frequency
+ * @param {number} classDuration
+ */
+function sendNewClientMessage(announcementId, availabilities, money, subject, level, frequency, classDuration) {
+
+	// Get the channel to which it will send the annoucements
+	const channel = client.channels.cache.get(mainChannelId);
+	// Create message object 
+	const msgEmbed = new EmbedBuilder()
+		.setColor(0x7289DA)
+		.setTitle('New Client Anouncement')
+		.setDescription(`**Subject:** ${subject} \n**Level:** ${level} \n**Class(es) per week:** ${frequency} \n**Pay per class:** ${money} \n**Time slots:** ${availabilities.join(", ")}\n**Class duration**: ${classDuration} hour(s)`)
+		.setTimestamp() 
+		.setFooter({ text: 'Please select the date and time that fits you best and we will get back to you on the next steps.', iconURL:'https://i.imgur.com/i1k870R.png'});
+
+	//Create a row object that will hold the select menu
+	const row = new ActionRowBuilder();
+
+	//Create Select Menu object
+	const menu = new SelectMenuBuilder()
+				.setCustomId("dateSelection")
+				.setPlaceholder(`Please select ${frequency} date option(s)`)
+				.setMinValues(frequency)
+				.setMaxValues(frequency)
+
+	//Loop through the availabilities' list and display every ability to tutor via select menu 
+	availabilities.forEach((dateAndTime) => {
+		menu.addOptions(
+			{
+				"label": dateAndTime,
+				"value": dateAndTime
+			}
+		)
+	});
+
+	//Add menu to row
+	row.addComponents(menu);
+
+	// Add submit button inside new row
+	const row2 = new ActionRowBuilder()
+		.addComponents( 
+			new ButtonBuilder()
+				.setCustomId("submitButton")
+				.setLabel('Submit request')
+				.setStyle(ButtonStyle.Success),
+
+			new ButtonBuilder()
+				.setCustomId("cancelButton")
+				.setLabel('Cancel request')
+				.setStyle(ButtonStyle.Danger),
+
+		);
+			
+	// Sends both objects to channel
+	channel.send({ embeds: [msgEmbed], components: [row,row2]});
+}
+
+
+client.on('interactionCreate', async interaction => {
+	
+	// The answers of all tutors that are stored inside the CSV in the form of an array
+	const answers = await csv().fromFile("answers.csv");
+
+	if (interaction.customId === 'dateSelection') {
+	
+		// Checks if the tutor has already an answer stored inside then CVS
+		if (tutorMadeASelection(interaction.user.id, answers)) {
+			interaction.reply({ content: "Please **submit** or **cancel** your first request to proceed.", ephemeral: true });
```

Comment:
Excellent logic, I would just change the text to "You have registered answers to a previous message which you have not yet submitted or canceled. Please do so before attempting to register new one for this message." I mean that text is not that good but the idea is to say something that explains a bit better bcs the request they left answers for may be far up in the channel and they have registered answers a long time ago